src/components/atoms/CopyButton/DownloadButton.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+import {Tooltip} from '@custom-antd';
+
+import {useDownloadFile} from '@hooks/useDownloadFile';
+
+import {StyledCopyOutlined} from './CopyButton.styled';
+
+type SaveOptions = {
+  content: string;
+  onClick?: () => void;
+  filename: string;
+};
+
+const DownloadButton: React.FC<SaveOptions> = props => {
+  const {content, onClick, filename} = props;
+  const {setProcessed} = useDownloadFile(content, {
+    filename,
+  });
+
+  const onDownloadClick = () => {
+    setProcessed(true);
+    onClick?.();
```

Comment:
I guess it basically replaces ```if (onClick) { onClick(); }```
Never seen this syntax

Reply:
This is the ES6 syntax for optional chaining: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

It will execute the function onClick if it is not undefined. Hence no additional check is required.

Reply:
This makes sense, never seen it on methods, but ok